open-notes/maturity-model.md

```diff
@@ -0,0 +1,394 @@
+
+### Telemetry capture
+
+Basic telemetry is captured and available in a central place for query. This includes things like CPU usage, disk usage, memory usage, request response times, etc.
+
+### Story readiness gate
+
+Stories are not even considered ready for work if there is any unclarity and have not gone through rigorous refinement.
+
+### Ubiquitous language
+
+Business and development use the same language to talk about software processes. These discussions happen in terms of business language and we should try not to force the business to adopt technical terms. When there is divergence, we should try to agree on common terminology. Code should reflect the ubiquitous language at all times.
+
+### Deploy weekly
+
+All applications should be deployed at least once a week. Even applications that are not changing should be deployed to validate the deployment pipeline.
+
+### Dashboards II
+
+From Dashboard I link to a dedicated dashboard the core telemetry data for that application such as memory, CPU, request count, response time, error rate. This is Dashboard II per application. Depending on space and number of applications owned you may want to add a more detailed health indicator like error rate alongside uptime on Dashboard I.
+
+### Uptime alerts
+
+Alerts should be setup to notify a team on a given channel if their application is unresponsive for a time. The time is dependent on SLO, based on criticality of that system, which should be found in default tags for that resource.
+
+### Secrets in key vault
+
+All secrets should be stored in a key vault with access following least privilege.
+
+### Custom metrics and KPIs
+
+The system sends custom metrics representing key business processes being fulfilled by the system. It must be possible to track and visualize KPIs based on these metrics.
+
+### Limited WIP
+
+By focusing on completing stories rather than maximizing work in-progress, we can add value as quickly as possible by focusing on the most valuable story. We achieve this by limiting work in progress (WIP) so development capacity -1. This means at all times someone is free to test, pair-program, handle support questions, or follow up on open questions. By helping others finish, they increase the delivery of value as well as help with knowledge share and social interaction.
+
+### Use-case entry points
+
+It is often useful when dealing with functionality that mutates state to wrap it up in it's own dedicated class that is not shared. This is serving a similar function that something like an application service is often serving but is more isolated. It makes dependencies per feature explicit and allows for easier testing per feature as only the dependencies for that feature are needed. Mostly, it serves as an entry point where the high-level story of how that feature works can be told. See [this post](https://devonburriss.me/managing-code-complexity/) for more discussion.
```

Comment:
You have generic skills (limit WIP), and then this tackles a very specific behaviour. What is the principle behind this skill? Is use-case entry points the only solution?